ddl/placement_policy.go

```diff
@@ -232,3 +241,53 @@ func onAlterPlacementPolicy(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	job.FinishDBJob(model.JobStateDone, model.StatePublic, ver, nil)
 	return ver, nil
 }
+
+func checkPlacementPolicyNotInUse(d *ddlCtx, t *meta.Meta, policy *model.PolicyInfo) error {
+	currVer, err := t.GetSchemaVersion()
+	if err != nil {
+		return err
+	}
+	is := d.infoCache.GetLatest()
+	if is.SchemaMetaVersion() == currVer {
+		return checkPlacementPolicyNotInUseFromInfoSchema(is, policy)
+	}
+
+	return checkPlacementPolicyNotInUseFromMeta(t, policy)
+}
+
+func checkPlacementPolicyNotInUseFromInfoSchema(is infoschema.InfoSchema, policy *model.PolicyInfo) error {
+	for _, dbInfo := range is.AllSchemas() {
```

Comment:
My first version is same as chaowang like here, concerning about this traveling loop will be costy, that's why I build a dependency cache in information schema @lcwangchao @xhebox

Reply:
Though, your thought is right, once ddl owner got a left-behind schema, it will cause some consistency problem, it's a trade-off here.

Reply:
I'm not with a strong requirement here, both is ok for me, just a notification here.

Reply:
I would prefer current implement because it is more easier. Further more, drop policy is a low frequency operation and it is not performance sensitive, keep it simple can help to reduce bugs.